.changeset/curly-turtles-yawn.md

```diff
@@ -0,0 +1,5 @@
+---
+'@itwin/itwinui-react': patch
+---
+
+Memoize `useToaster` return value so it can be more reliably used in dependency arrays.
```

Comment:
nit (optional): Not sure if this is the correct grammar or just my personal preference.

```suggestion
Memoize `useToaster`'s return value so that it can be more reliably used in dependency arrays.
```